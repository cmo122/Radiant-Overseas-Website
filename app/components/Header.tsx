import { rem,Text, Container, useMantineColorScheme, Button, Group, Burger, Image, Center, Drawer, Flex } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import ROELOGO from '../assets/logos/ROELINELOGO.png'
import { IconSun, IconMoon } from '@tabler/icons-react';
import { scroller } from 'react-scroll';
import cx from 'clsx';
import classes from '../css/HeaderSimple.module.css';

const links = [
  { link: 'carriersSection', label: 'Our Carriers' },
  { link: 'expertiseSection', label: 'Our Expertise' },
  { link: 'contactForm', label: 'Contact Us' },
];

export default function Header() {
  const [opened, { toggle: toggleDrawer, close: closeDrawer }] = useDisclosure(false);

  const scrollOptions={
    duration: 400, 
    smooth: true,
    spy: true,
  }

  const headerItems = links.map((link, index) => (
    <Button key={index} m="sm" className={classes.link}
    id={`${link.label} Button`}
    fw="900"
    variant="gradient"
    gradient={{ from: 'orange', to: 'yellow' }}
    onClick={() => {
      closeDrawer();
      const elementId = link.link; 
      scroller.scrollTo(elementId, scrollOptions);
      
    }}>
        <Text>
          {link.label}
        </Text>
    </Button>
  ));
  
  const changeThemeButton= 
  <Button id="changeThemeButton" name="changeThemeButton" key="4" 
    onClick={() => {
    closeDrawer();
    toggleColorScheme()}
    }
    className={classes.link}
    m="sm"
    variant="gradient"
    gradient={{ from: 'orange', to: 'yellow', deg: 90 }}>
    <IconSun className={cx(classes.icon, classes.light)} stroke={1.5} />
    <IconMoon className={cx(classes.icon, classes.dark)} stroke={1.5} />
  </Button>

  const items=[...headerItems,changeThemeButton]

  const { toggleColorScheme } = useMantineColorScheme();

  return (
      <header className={classes.header} role="header">
        <Container size="lg" className={classes.inner} >
          <Group>
            <Image src={ROELOGO.src} alt="ROE LINE logo" fit="contain" w="100" h="100" data-testid="logo"/>
          </Group>
          <Center  visibleFrom="sm">
            {items}
          </Center>
          <Burger opened={opened} onClick={toggleDrawer} hiddenFrom="sm" size="lg" role="burgermenu"
          id="burgermenu"/>
          <Drawer opened={opened} onClose={closeDrawer } ta="center"
          id="drawer"
          styles={{close:{'width':rem(50),'height':rem(50)}}}
          closeButtonProps={{ 'aria-label': 'close' }}>
            <Flex direction="column" justify="center" align="center">
              {items}
            </Flex>
          </Drawer>
        </Container>
      </header>
  );
}