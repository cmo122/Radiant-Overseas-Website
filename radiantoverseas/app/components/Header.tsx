import { useState } from 'react';
import { Container, useMantineColorScheme, Button, Group, Burger } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import {IconPackage} from '@tabler/icons-react'
import { IconSun, IconMoon } from '@tabler/icons-react';
import cx from 'clsx';
import classes from '../css/HeaderSimple.module.css';

const links = [
  { link: '/about', label: 'About Us' },
  { link: '/carriers', label: 'Our Carriers' },
];

export default function Header() {
  const [opened, { toggle }] = useDisclosure(false);
  const [active, setActive] = useState(links[0].link);

  const items = links.map((link) => (
    <a
      key={link.label}
      href={link.link}
      className={classes.link}
      data-active={active === link.link || undefined}
      onClick={(event) => {
        event.preventDefault();
        setActive(link.link);
      }}
    >
      {link.label}
    </a>
  ));
    
  const { toggleColorScheme } = useMantineColorScheme();

  return (
    <header className={classes.header}>
      <Container size="md" className={classes.inner}>
        <Group>
          <IconPackage size={40} />
          <h3 style={{
                    color: 'linear-gradient(to right, yellow, blue, orange)',
                    fontFamily: 'Futura, sans-serif',
                    fontSize: '2rem',
                    fontWeight: 'bold',
                  }}>Radiant Overseas Express</h3>
        </Group>
        <Group gap={5} visibleFrom="xs">
          {items}
          <Button onClick={() => toggleColorScheme()}>
          <IconSun className={cx(classes.icon, classes.light)} stroke={1.5} />
          <IconMoon className={cx(classes.icon, classes.dark)} stroke={1.5} />
          </Button>
        </Group>

        <Burger opened={opened} onClick={toggle} hiddenFrom="xs" size="sm" />
      </Container>
    </header>
  );
}