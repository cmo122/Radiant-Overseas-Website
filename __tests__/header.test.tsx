// Header.test.js

import React from 'react';
import { render, screen } from '../test-utils';
import Header from '@/app/components/Header';

beforeEach(() => {
  const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null
  });
  window.IntersectionObserver = mockIntersectionObserver;
  render(<Header />)
});



describe('Header Component testing', () => {
    describe('Desktop Header',()=>{
        it('renders navigation links', () => {
            const carrierLink = screen.getByText('Our Carriers');
            const expertiseLink = screen.getByText('Our Expertise');
            const contactLink = screen.getByText('Contact Us');

            expect(carrierLink).toBeInTheDocument();
            expect(expertiseLink).toBeInTheDocument();
            expect(contactLink).toBeInTheDocument();
        });

        it('renders the logo', () => {
            const logo = screen.getByTestId('logo');
            expect(logo).toBeInTheDocument();
        });
    })
});