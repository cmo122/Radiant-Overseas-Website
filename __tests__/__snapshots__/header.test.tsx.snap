// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component testing Desktop Header matches the snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <header
    class="header"
    role="header"
  >
    <div
      class="inner m-7485cace mantine-Container-root"
      data-size="md"
      style="--container-size: var(--container-size-md);"
    >
      <div
        class="m-4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <img
          alt="ROE LINE logo"
          class="m-9e117634 mantine-Image-root"
          data-testid="logo"
          src="/img.jpg"
          style="--image-object-fit: contain;"
        />
      </div>
      <div
        class="m-4451eb3a mantine-Center-root mantine-visible-from-sm"
      >
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Carriers Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Carriers
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Expertise Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Expertise
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Contact Us Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Contact Us
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="changeThemeButton"
          name="changeThemeButton"
          style="--button-bg: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <svg
                class="icon light"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                />
                <path
                  d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                />
              </svg>
              <svg
                class="icon dark"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <button
        class="mantine-focus-auto m-fea6bf1a mantine-Burger-root m-87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
        data-size="lg"
        style="--burger-size: var(--burger-size-lg);"
        type="button"
      >
        <div
          class="m-d4fb9cad mantine-Burger-burger"
          data-reduce-motion="true"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header Component testing matches the snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <header
    class="header"
    role="header"
  >
    <div
      class="inner m-7485cace mantine-Container-root"
      data-size="md"
      style="--container-size: var(--container-size-md);"
    >
      <div
        class="m-4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <img
          alt="ROE LINE logo"
          class="m-9e117634 mantine-Image-root"
          data-testid="logo"
          src="/img.jpg"
          style="--image-object-fit: contain;"
        />
      </div>
      <div
        class="m-4451eb3a mantine-Center-root mantine-visible-from-sm"
      >
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Carriers Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Carriers
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Expertise Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Expertise
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Contact Us Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Contact Us
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="changeThemeButton"
          name="changeThemeButton"
          style="--button-bg: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <svg
                class="icon light"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                />
                <path
                  d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                />
              </svg>
              <svg
                class="icon dark"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <button
        class="mantine-focus-auto m-fea6bf1a mantine-Burger-root m-87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
        data-size="lg"
        style="--burger-size: var(--burger-size-lg);"
        type="button"
      >
        <div
          class="m-d4fb9cad mantine-Burger-burger"
          data-reduce-motion="true"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <header
    class="header"
    role="header"
  >
    <div
      class="inner m-7485cace mantine-Container-root"
      data-size="md"
      style="--container-size: var(--container-size-md);"
    >
      <div
        class="m-4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <img
          alt="ROE LINE logo"
          class="m-9e117634 mantine-Image-root"
          data-testid="logo"
          src="/img.jpg"
          style="--image-object-fit: contain;"
        />
      </div>
      <div
        class="m-4451eb3a mantine-Center-root mantine-visible-from-sm"
      >
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Carriers Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Carriers
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Our Expertise Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Our Expertise
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="Contact Us Button"
          style="--button-bg: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(45deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none; font-weight: 900;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <p
                class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
              >
                Contact Us
              </p>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active link m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="gradient"
          id="changeThemeButton"
          name="changeThemeButton"
          style="--button-bg: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-hover: linear-gradient(90deg, var(--mantine-color-orange-filled) 0%, var(--mantine-color-yellow-filled) 100%); --button-color: var(--mantine-color-white); --button-bd: none;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-811560b9 mantine-Button-label"
            >
              <svg
                class="icon light"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
                />
                <path
                  d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
                />
              </svg>
              <svg
                class="icon dark"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <button
        class="mantine-focus-auto m-fea6bf1a mantine-Burger-root m-87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
        data-size="lg"
        style="--burger-size: var(--burger-size-lg);"
        type="button"
      >
        <div
          class="m-d4fb9cad mantine-Burger-burger"
          data-reduce-motion="true"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
